### DISCUSSION TITLE
Changing mapping between file name and url path

### DISCUSSION THREAD

**Discussion by iAdramelk** [vercel/next.js#21902] [Help]
I'm building docs site using next.js and mdx. Docs have current version and old versions.

For current version I want to have urls like `/docs/installation`. For older versions I want to have urls like `/docs/v/1.0/installation`, but I don't want to move files between folders every time new version is released, I want to have file structure like this:

```
pages/
  v/
   1.0/
     installation.mdx
   2.0/
     installation.mdx
```

But serve latest version docs from the paths without `/v/x.x/` prefix.

I can't use dynamic routes here because of mdx (it can't be serialized to the json) and I want to use serverless functions and API routes in the future, so `exportPathMap` didn't work for me too.

Is it possible to do something like it with Next?

**Comment by michrome** [comment]
Hi @iAdramelk have you looked at using [`rewrites`](https://nextjs.org/docs/api-reference/next.config.js/rewrites)?

To start with, your `next.config.js` might look like the below.

```js
const LATEST_VERSION = "2.0";

module.exports = {
  async rewrites() {
    return [
      {
        source: "/docs/:slug",
        destination: `/v/${LATEST_VERSION}/:slug`,
      },
    ]
  },
}
```

**Reply by iAdramelk** [reply]
Hi @michrome , it indeed worked, thanks! I tried it before, but probably misconfigure it somehow and wasn't able to achieve result I wanted. Now I tried it again and it worked.

**Reply by iAdramelk** [reply]
@michrome tried it once more and it looks like it still didn't work. :( It work then I load the page from the server, but on client navigation it redirects me to the original route.

E. g. load page `/docs/quickstart/` - works. But clicking link `/docs/installation/` inside the page will render page with the url `/docs/v/5.0/installation/`.

**Reply by iAdramelk** [reply]
Ok. It's actually more strange than that:

- In 10.0.5 dev mode it works as I described above.
- In 10.0.6 dev mode navigates as it should, bit then you hover the link, tooltip shows the path with the version in it.
- In 10.0.6 build everything looks and works as it should.

So it is probably some bug.
